tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/SyncMethodInvokerIntegration.cs

```diff
@@ -52,8 +50,15 @@ public static CallTargetState OnMethodBegin<TTarget>(TTarget instance, object in
                 return CallTargetState.GetDefault();
             }
 
-            var requestContext = WcfCommon.GetCurrentOperationContext()?.GetProperty<object>("RequestContext").GetValueOrDefault();
-            return new CallTargetState(WcfCommon.CreateScope(requestContext));
+            var operationContext = WcfCommon.GetCurrentOperationContext();
+            if (operationContext != null && operationContext.TryDuckCast<IOperationContextStruct>(out var operationContextProxy))
+            {
+                return new CallTargetState(WcfCommon.CreateScope(operationContextProxy.RequestContext));
+            }
+            else
+            {
+                return CallTargetState.GetDefault();
+            }
```

Comment:
[nit] This whole body could be moved in WcfCommon and used here and in AsyncMethodInvoker_InvokeBegin_Integration.cs and TaskMethodInvokerIntegration.cs (even the TODOs are duplicated :))

Reply:
Good catch! Maybe in a separate PR since it's not related to removing `Datadog.Trace.ClrProfiler.Emit`?